src/cargo/core/package_id_spec.rs

```diff
@@ -17,16 +17,16 @@ pub struct PackageIdSpec {
 impl PackageIdSpec {
     pub fn parse(spec: &str) -> CargoResult<PackageIdSpec> {
         if spec.contains("/") {
-            match spec.to_url() {
-                Ok(url) => return PackageIdSpec::from_url(url),
-                Err(..) => {}
-            }
             if !spec.contains("://") {
```

Comment:
I had to move this up so that support for auto-inserting `cargo://` runs before the attempt to parse the URL.

The `good_parsing()` test (in `src/cargo/core/package_id_spec.rs`) catches this:

```rust
        ok("crates.io/foo", PackageIdSpec {
            name: "foo".to_string(),
            version: None,
            url: Some(url("cargo://crates.io/foo").unwrap()),
        });
```

Without this change, "crates.io/foo" would be parsed as a relative path to the current directory.